ibm/data_source_ibm_iam_api_key.go

```diff
@@ -0,0 +1,152 @@
+/**
+ * (C) Copyright IBM Corp. 2021.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ibm
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM/platform-services-go-sdk/iamidentityv1"
+)
+
+func dataSourceIbmIamApiKey() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIbmIamApiKeyRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": &schema.Schema{
+				Type:        schema.TypeString,
```

Comment:
We shdn't have id as argument...I think take the name from the user and fetch the required key